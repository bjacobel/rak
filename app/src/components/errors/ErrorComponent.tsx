import { styled } from '@linaria/react';

import { SHOW_STACK } from '../../constants';
import { robotoMono } from '../../stylesheets/shared.css';

const MonoHeader = styled.h3`
  ${robotoMono}
`;

const ErrorComponent = ({ error }: { error?: Error }) => (
  <>
    <MonoHeader>Something went wrong.</MonoHeader>
    {SHOW_STACK && error && error.stack && <pre data-testid="stack">{error.stack}</pre>}
  </>
);

export default ErrorComponent;
