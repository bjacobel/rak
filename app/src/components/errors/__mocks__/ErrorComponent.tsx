export default () => (
  <>
    <h3 className="errorHeader">Something went wrong.</h3>
    <pre>In the real component, the stack would be rendered here.</pre>
  </>
);
