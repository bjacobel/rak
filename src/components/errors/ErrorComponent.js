import React from 'react';

import { SHOW_STACK } from 'constants/index';
import { errorHeader } from './style.css';

export default ({ error }) => (
  <>
    <h3 className={errorHeader}>Something went wrong.</h3>
    {SHOW_STACK && error && error.stack && <pre data-testid="stack">{error.stack}</pre>}
  </>
);
