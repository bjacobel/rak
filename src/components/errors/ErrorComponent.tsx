import React from 'react';

import { SHOW_STACK } from '../../constants';
import { errorHeader } from './style.css';

const ErrorComponent = ({ error }: { error?: Error }) => (
  <>
    <h3 className={errorHeader}>Something went wrong.</h3>
    {SHOW_STACK && error && error.stack && <pre data-testid="stack">{error.stack}</pre>}
  </>
);

ErrorComponent.defaultProps = {
  error: undefined,
};

export default ErrorComponent;
