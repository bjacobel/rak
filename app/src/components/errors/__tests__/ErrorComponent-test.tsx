import { render } from '../../../testing/rtl';
import ErrorComponent from '../ErrorComponent';

const mockShowStackConstant = jest.fn().mockReturnValue(true);

jest.mock('../../../constants', () => ({
  get SHOW_STACK() {
    return mockShowStackConstant();
  },
}));

describe('error rendering component', () => {
  describe('in dev mode', () => {
    it('matches snapshot without an error', () => {
      const { asFragment } = render(<ErrorComponent />);

      expect(asFragment()).toMatchSnapshot();
    });

    it('has error string in document', () => {
      const errString = 'idk lol';
      const { getByTestId } = render(<ErrorComponent error={new Error(errString)} />);

      expect(getByTestId('stack').innerHTML).toMatch(`Error: ${errString}`);
    });
  });

  describe('in prod mode', () => {
    it('does not print stack even with an error', () => {
      mockShowStackConstant.mockReturnValueOnce(true);

      const errString = 'idk lol';
      const { queryByText } = render(<ErrorComponent error={new Error(errString)} />);

      expect(queryByText(errString)).not.toBeInTheDocument();
    });
  });
});
