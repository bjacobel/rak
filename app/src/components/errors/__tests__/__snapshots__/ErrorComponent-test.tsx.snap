// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`error rendering component in dev mode matches snapshot without an error 1`] = `
<DocumentFragment>
  <h3>
    Something went wrong.
  </h3>
</DocumentFragment>
`;
